src/realm/list.cpp

```diff
@@ -371,6 +371,16 @@ Obj LnkLst::create_and_insert_linked_object(size_t ndx)
 Obj LnkLst::create_and_set_linked_object(size_t ndx)
 {
     Table& t = *get_target_table();
+    if (t.is_embedded() && m_list.get(ndx) != ObjKey{}) {
```

Comment:
This approach looks good to me. The other option would be to just clear out the existing contents of the object here and then just update it with the new object, but not sure if that is possible